packages/ui/Carousel/CarouselDrag.ts

```diff
@@ -0,0 +1,93 @@
+import type { BaseConfig, BaseProps, DragServiceProps } from '@studiometa/js-toolkit';
+import { withDrag, withMountOnMediaQuery } from '@studiometa/js-toolkit';
+import { inertiaFinalValue } from '@studiometa/js-toolkit/utils';
+import { AbstractCarouselChild } from './AbstractCarouselChild.js';
+import { getClosestIndex } from './utils.js';
+
+/**
+ * Props for the CarouselDrag class.
+ */
+export interface CarouselDragProps extends BaseProps {}
+
+/**
+ * CarouselDrag class.
+ */
+export class CarouselDrag<
+  T extends BaseProps = BaseProps,
+> extends withMountOnMediaQuery<AbstractCarouselChild>(
+  withDrag(AbstractCarouselChild),
+  '(pointer: fine)',
+)<T & CarouselDragProps> {
+  /**
+   * Config.
+   */
+  static config: BaseConfig = {
+    name: 'CarouselDrag',
+  };
+
+  /**
+   * Dragged hook.
+   */
+  dragged(props: DragServiceProps) {
+    if (!this.$isMounted) return;
+
+    // do noting on inertia and stop
+    if (props.mode === 'inertia' || props.mode === 'stop') {
+      return;
+    }
+
+    // do nothin while the distance is 0
+    if (
+      (this.isHorizontal && props.distance.x === 0) ||
+      (this.isVertical && props.distance.y === 0)
+    ) {
+      return;
+    }
+
+    const wrapper = this.$el;
+
+    // @todo wait for the props.delta values to be fixed
+    // @see https://github.com/studiometa/js-toolkit/pull/533
```

Comment:
@titouanmathis is it ok now ? Do I need to edit something according to the merge of !533 ?

Reply:
https://github.com/studiometa/js-toolkit/pull/533 has been merged and published in v3.0.0, so the `@todo` can be removed I think